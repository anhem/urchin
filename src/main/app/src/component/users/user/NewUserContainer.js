import React, {Component} from "react";
import {connect} from "react-redux";
import NewUser from "./NewUser";
import {createUser, setUser} from "../../../action/userAction";

class NewUserContainer extends Component {

    render() {
        return (
            <NewUser
                user={this.props.user}
                fieldErrors={this.props.fieldErrors}
                callbacks={{
                    setUser: this.props.setUser,
                    createUser: this.props.createUser
                }}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setUser: (user) => {
            dispatch(setUser(user))
        },
        createUser: (user) => {
            dispatch(createUser(user))
        }
    }
};

const mapStateToProps = (state) => {
    return {
        user: state.userData.user || {},
        fieldErrors: state.userData.fieldErrors || {}
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(NewUserContainer)