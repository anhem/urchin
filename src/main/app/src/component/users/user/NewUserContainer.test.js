import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {NewUserContainer} from "./NewUserContainer";

describe('NewUserContainer', () => {

    let props = {
        user: {},
    };

    it('renders without crashing', () => {
        expect(mount(<NewUserContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<NewUserContainer {...props}/>))).toMatchSnapshot();
    });
});
