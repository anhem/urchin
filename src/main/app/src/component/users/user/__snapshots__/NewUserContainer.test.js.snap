// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUserContainer match snapshot 1`] = `
<NewUser
  callbacks={
    Object {
      "createUser": undefined,
      "setUser": undefined,
    }
  }
  user={Object {}}
/>
`;
